stdlib/public/Python/Python.swift

```diff
@@ -1227,29 +1232,21 @@ extension PythonObject : Hashable {
   }
 }
 
-extension PythonObject : MutableCollection {
-  public typealias Index = PythonObject
-  public typealias Element = PythonObject
-
-  public var startIndex: Index {
-    return 0
-  }
-
-  public var endIndex: Index {
-    return Python.len(self)
-  }
-
-  public func index(after i: Index) -> Index {
-    return i + PythonObject(1)
-  }
-
-  public subscript(index: PythonObject) -> PythonObject {
-    get {
-      return self[index as PythonConvertible]
-    }
-    set {
-      self[index as PythonConvertible] = newValue
-    }
+extension PythonObject: IteratorProtocol, Sequence {
+
+  public func next() -> PythonObject? {
```

Comment:
Hmm, this is looking simpler: 
  guard let result: OwnedPyObjectPointer = PyIter_Next(obj.borrowedPyObject) else {
      try! throwPythonErrorIfPresent()
      return nil
    }
    return PythonObject(owning: result)

Reply:
This is exactly what I had initially, but I refrained from the straight C-API calls.